### PR TITLE
chore: Deprecate eval itests

### PR DESCRIPTION
**Pull request by HasanAlrimawi**
This PR serves as a part of #22907 .

### CODE REVIEW

File: tests/testdata/run/lock_check_ok2.json [hunk 1]
```diff
@@ -9,5 +9,20 @@
...
+        "npm:@jsr/denotest__add@1",
+        "npm:@jsr/std__assert@1",
+        "npm:@jsr/std__http@1",
+        "npm:@ngrx/store-devtools@^14.3.2",
+        "npm:@stripe/stripe-js@^2.1.6",
+        "npm:chalk@^4.1.2",
+        "npm:path-to-regexp@7",
+        "npm:stripe@^13.8.0",
+        "npm:uuid@^3.4.0"
+      ]
```
Comment (dsherret) [inline tests/testdata/run/lock_check_ok2.json:25]:
```suggestion
  }
```

File: tests/testdata/run/lock_v2_check_ok2.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "2",
+  "version": "4",
```
Comment (dsherret) [inline tests/testdata/run/lock_v2_check_ok2.json:2]:
```suggestion
  "version": "2",
```

File: tests/testdata/run/lock_v2_check_ok2.json [hunk 3]
```diff
@@ -9,5 +9,20 @@
...
+        "npm:@jsr/std__assert@1",
+        "npm:@jsr/std__http@1",
+        "npm:@ngrx/store-devtools@^14.3.2",
+        "npm:@stripe/stripe-js@^2.1.6",
+        "npm:chalk@^4.1.2",
+        "npm:path-to-regexp@7",
+        "npm:stripe@^13.8.0",
+        "npm:uuid@^3.4.0"
+      ]
+    }
```
Comment (dsherret) [inline tests/testdata/run/lock_v2_check_ok2.json:26]:
```suggestion
  }
```

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

File: tests/integration/eval_tests.rs [hunk 4]
```diff
@@ -43,48 +25,3 @@ fn eval_randomness() {
   numbers.dedup();
   assert!(numbers.len() > 1);
 }
-
-itest!(eval_basic {
-  args: "eval console.log(\"hello\")",
-  output_str: Some("hello\n"),
-});
```
Comment (dsherret) [inline tests/integration/eval_tests.rs:50]: I think ./tools/lint.js needs to be updated to have the new number though. Probably linting will fail on the ci

**Comment by HasanAlrimawi** [comment]
The test specs::eval::env_file keeps failing on the CI checks but it always succeeds on local runs.
What could the issue be? 
@dsherret

File: tests/specs/eval/env_file/__test__.jsonc [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "args": "eval --env-file=env_file console.log(Deno.env.get(\"Another_FOO\"));",
```
Comment (dsherret) [inline tests/specs/eval/env_file/__test__.jsonc:2]:
```suggestion
  "args": "eval --env-file=env_file console.log(Deno.env.get(\"ANOTHER_FOO\"));",
```

Comment (dsherret) [reply tests/specs/eval/env_file/__test__.jsonc]: @HasanAlrimawi this was the fix. It's case sensitive on mac and linux

File: tests/specs/eval/env_file_missing/__test__.jsonc [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "args": "eval --env-file=missing console.log(Deno.env.get(\"Another_FOO\"));",
```
Comment (marvinhagemeister) [inline tests/specs/eval/env_file_missing/__test__.jsonc:2]:
```suggestion
  "args": "eval --env-file=missing console.log(Deno.env.get(\"ANOTHER_FOO\"));",
```

File: tests/specs/eval/env_unparsable_file/__test__.jsonc [hunk 7]
```diff
@@ -1,4 +1,4 @@
 {
-  "args": "eval --env=../../../testdata/env_unparsable console.log(Deno.env.get(\"Another_FOO\"))",
+  "args": "eval --env-file=env_unparsable console.log(Deno.env.get(\"Another_FOO\"));",
```
Comment (marvinhagemeister) [inline tests/specs/eval/env_unparsable_file/__test__.jsonc:2]:
```suggestion
  "args": "eval --env-file=env_unparsable console.log(Deno.env.get(\"ANOTHER_FOO\"));",
```

**Pull request merged**